src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -102,6 +102,21 @@ def exit_code(self):
     return self._exit_code
 
 
+class _PantsProductPrecomputeFailed(Exception):
```

Comment:
So, at a fundamental level, there is no need to "precompute" or "warm" anything anymore. So we should actually be able to delete the warming codepath entirely.

Reply:
Added a TODO :)